components/director/pkg/webhook/types.go

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright 2020 The Compass Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package webhook
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"text/template"
+
+	"github.com/pkg/errors"
+)
+
+type Mode string
+
+const (
+	ModeSync  Mode = "SYNC"
+	ModeAsync Mode = "ASYNC"
+)
+
+// Resource is used to identify entities which can be part of a webhook's request data
+type Resource interface {
+	Sentinel()
```

Comment:
Try removing this method. I think it is not needed as every struct will implicitly implement it. As the method is not used it should be safe to remove.

Reply:
I know we discussed it and I couldn't remember why we wanted to have some dummy method, but now I remember - removing the dummy method would practically make any type implicitly implement this interface. This would be just an indirect way of using `interface{}`. So the idea behind the dummy method was to at least restrict to some extent the types which implement this so that we can limit any potential problems later on.